src/isl/util.rs

```diff
@@ -270,11 +276,16 @@ impl Range {
 
     // helper method to which validates a non negative integer range boundary value
     pub fn validate_non_negative_integer_range_boundary_value(
-        value: &AnyInt,
+        value: &IntegerValue,
+        range_type: &RangeType,
     ) -> IonSchemaResult<usize> {
         match value.as_i64() {
             Some(v) => {
-                if v >= 0 {
+                // minimum precision must be greater than or equal to 1
+                // for more information: https://amzn.github.io/ion-schema/docs/spec.html#precision
+                if (range_type == &RangeType::PrecisionRange && v >= 1)
+                    || (range_type != &RangeType::PrecisionRange && v >= 0)
```

Comment:
Cod re-use is good, but do the `Err` messages on lines 291, 298, 313, and 320 also need to be updated?